pkg/test/e2e/nodeport-proxy/network.go

```diff
@@ -125,12 +136,17 @@ func (n *NetworkingTestConfig) CleanUp() error {
 // maxTries == minTries will confirm that we see the expected endpoints and no
 // more for maxTries. Use this if you want to eg: fail a readiness check on a
 // pod and confirm it doesn't show up as an endpoint.
-func (n *NetworkingTestConfig) DialFromNode(targetIP string, targetPort, maxTries, minTries int, expectedEps sets.String) sets.String {
+func (n *NetworkingTestConfig) DialFromNode(targetIP string, targetPort, maxTries, minTries int, expectedEps sets.String, https bool) sets.String {
 	ipPort := net.JoinHostPort(targetIP, strconv.Itoa(targetPort))
 	// The current versions of curl included in CentOS and RHEL distros
 	// misinterpret square brackets around IPv6 as globbing, so use the -g
 	// argument to disable globbing to handle the IPv6 case.
-	cmd := fmt.Sprintf("curl -g -q -s --max-time 15 --connect-timeout 1 http://%s/hostName", ipPort)
+	cmd := ""
+	if n.Https == true {
```

Comment:
We also need to handle the case where we specify the hostname to test the HTTP2 CONNECT entry point